src/nvim/ex_docmd.c

```diff
@@ -3658,6 +3658,26 @@ const char * set_one_cmd_context(
     xp->xp_pattern = (char_u *)arg;
     break;
 
+  case CMD_lua:
+    xp->xp_context = EXPAND_LUA;
+    // Drop print(, vim.inspect(, and vim. from xp->xp_pattern
+    if (STRLEN(xp->xp_pattern) >= 7) {
+      if (memcmp(xp->xp_pattern, "print(", 6) == 0) {
+        xp->xp_pattern += 6;
+      }
+    }
+    if (STRLEN(xp->xp_pattern) >= 13) {
+      if (memcmp(xp->xp_pattern, "vim.inspect(", 12) == 0) {
+        xp->xp_pattern += 12;
+      }
+    }
+    if (STRLEN(xp->xp_pattern) >= 5) {
+      if (memcmp(xp->xp_pattern, "vim.", 4) == 0) {
+        xp->xp_pattern += 4;
+      }
+    }
```

Comment:
I should extract this logic out into a function, but I don't know where to put it.

Reply:
This Logic already exists in the lua code (and is more general), the question is how to get it into c which didn't work in tjs PR.